db/migrate/20130329192605_create_tools.rb

```diff
@@ -10,6 +10,6 @@ def change
       t.timestamps
     end
 
-    add_index :tools, :barcode
+    add_index :tools, :barcode, unique: true
```

Comment:
Can you move the unique flag to the actual barcode field on line 5 rather than the index?
 Add a line note